core/domain/classifier_domain.py

```diff
@@ -154,7 +154,8 @@ def validate(self):
         utils.require_valid_name(
             self.algorithm_id, 'the algorithm id')
         if self.algorithm_id not in (
-                feconf.INTERACTION_CLASSIFIER_MAPPING.values()):
+                feconf.INTERACTION_CLASSIFIER_MAPPING[
+                    'TextInput']['classifier_id']):
```

Comment:
classifier_domain is a general class and should not be referring to specific interaction IDs. Write this to be robust against the future case where we support multiple algorithms, so that we don't have to come back to it later.